types/bull/index.d.ts

```diff
@@ -171,6 +171,8 @@ declare namespace Bull {
 
     returnvalue: any;
 
+    failedReason: string;
```

Comment:
It can still be undefined if the job doesn't fail.

Reply:
So should it be changed to be this?
```suggestion
    failedReason: string | undefined;
```

Reply:
Yes, or
```suggestion
    failedReason?: string;
```

Reply:
I always forget about the ? optional property. It looks a lot cleaner 😆